sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -526,95 +513,4 @@ trait Row extends Serializable {
   private def getAnyValAs[T <: AnyVal](i: Int): T =
     if (isNullAt(i)) throw QueryExecutionErrors.valueIsNullError(i)
     else getAs[T](i)
-
-  /**
-   * The compact JSON representation of this row.
-   * @since 3.0
-   */
-  @Unstable
-  def json: String = compact(jsonValue)
```

Comment:
although this is an unstable API, can we still add a migration guide and say that people can use `ToJsonUtil` instead?

Reply:
@cloud-fan is there an example of adding migration guide?

Reply:
Should add an item at https://github.com/apache/spark/blob/master/docs/sql-migration-guide.md

Reply:
Actually I think we shouldn't remove this ... we can deprecate them.

Reply:
added migration guide and also mark both methods as deprecated.

Reply:
@HyukjinKwon I think it is OK to remove since it is unstable. My main motivation for removing it is that it will significantly reduce the amount of code we need to move to the sql/api module, and that it needs SQLConfs to properly function.

Reply:
👌